src/thunderbots/software/ai/navigator/path_planning_navigator/path_planning_navigator.cpp

```diff
@@ -0,0 +1,126 @@
+{
+    this->world = world;
+
+    auto assigned_primitives = std::vector<std::unique_ptr<Primitive>>();
+    for (const auto &intent : assignedIntents)
+    {
+        intent->accept(*this);
+        assigned_primitives.emplace_back(std::move(current_primitive));
+    }
+
+    return assigned_primitives;
+}
+
+void PathPlanningNavigator::visit(const CatchIntent &catch_intent)
+{
+    auto p            = std::make_unique<CatchPrimitive>(catch_intent);
+    current_primitive = std::move(p);
+}
+
+void PathPlanningNavigator::visit(const ChipIntent &chip_intent)
+{
+    auto p            = std::make_unique<ChipPrimitive>(chip_intent);
+    current_primitive = std::move(p);
+}
+
+void PathPlanningNavigator::visit(const DirectVelocityIntent &direct_velocity_intent)
+{
+    auto p            = std::make_unique<DirectVelocityPrimitive>(direct_velocity_intent);
+    current_primitive = std::move(p);
+}
+
+void PathPlanningNavigator::visit(const DirectWheelsIntent &direct_wheels_intent)
+{
+    auto p            = std::make_unique<DirectWheelsPrimitive>(direct_wheels_intent);
+    current_primitive = std::move(p);
+}
+
+void PathPlanningNavigator::visit(const DribbleIntent &dribble_intent)
+{
+    auto p            = std::make_unique<DribblePrimitive>(dribble_intent);
+    current_primitive = std::move(p);
+}
+
+void PathPlanningNavigator::visit(const KickIntent &kick_intent)
+{
+    auto p            = std::make_unique<KickPrimitive>(kick_intent);
+    current_primitive = std::move(p);
+}
+
+double PathPlanningNavigator::place_holder_violation_func(const Point &input_point)
+{
+    return 0.0;
+}
+
+void PathPlanningNavigator::visit(const MoveIntent &move_intent)
+{
+    std::vector<Obstacle> obstacles = {};
+
+    for (auto &robot : world.enemyTeam().getAllRobots())
+    {
+        Obstacle o = Obstacle::createRobotObstacleWithScalingParams(robot, 1.2, 0);
+        obstacles.push_back(o);
+    }
+
+    for (auto &robot : world.friendlyTeam().getAllRobots())
+    {
+        if (robot.id() == move_intent.getRobotId())
+        {
+            // skip current robot
+            continue;
+        }
+        Obstacle o = Obstacle::createRobotObstacleWithScalingParams(robot, 1.2, 0);
+        obstacles.push_back(o);
+    }
+
+    auto p            = std::make_unique<MovePrimitive>(move_intent);
+    auto path_planner = std::make_unique<ThetaStarPathPlanner>(
+        this->world.field(), this->world.ball(), obstacles);
+
+    auto path_points = path_planner->findPath(
+        this->world.friendlyTeam().getRobotById(p->getRobotId())->position(),
+        p->getDestination());
+
+    if (path_points)
+    {
+        auto next_point = (*path_points)[1];
+        auto move       = std::make_unique<MovePrimitive>(
+            p->getRobotId(), next_point, move_intent.getFinalAngle(),
+            calculateTransitionSpeedBetweenSegments((*path_points)[0], (*path_points)[1],
+                                                    (*path_points)[2], 0),
```

Comment:
We need to make sure we're copying over the `enable_autokick` and `enable_dribbler` parameters from the `MoveIntent` here. It might be better to copy the `MovePrimitive` called `p` you made above, and then modify the target position/speed/angle on that (you might need to add some setters to the `MovePrimitive` class to do that). This would ensure any flags we add in the future get properly copied over as well.